ImperatorToCK3/Source/CK3/Titles/Title.cpp

```diff
@@ -10,6 +9,35 @@
 #include "ParserHelpers.h"
 
 
+
+void CK3::Title::addFoundTitle(const std::shared_ptr<Title>& newTitle, std::map<std::string, std::shared_ptr<Title>>& foundTitles)
+{
+	for (const auto& [locatedTitleName, locatedTitle] : newTitle->foundTitles)
+	{
+		if (newTitle->titleName.starts_with("c_")) // has county prefix = is a county
+		{
+			auto baronyProvince = locatedTitle->getProvince();
+			if (baronyProvince)
+			{
+				if (locatedTitleName == newTitle->capitalBarony)
+				{
+					newTitle->capitalBaronyProvince = *baronyProvince;
+				}
+				newTitle->addCountyProvince(*baronyProvince); // add found baronies' provinces to countyProvinces
+			}
+		}
+		foundTitles[locatedTitleName] = locatedTitle;
```

Comment:
You should be able to avoid multiple lookups with something like:
```
auto foundTitle = foundTitles[locatedTitleName];
foundTitle = locatedTitle;
if (!foundTitle->getDeJureLiege())
	foundTitle->setDeJureLiege(newTitle);
```
Though I'm a little unsure on the exact types, so it could be different.

An alternative could be to use
```
foundTitles.insert_or_assign() 
```

Reply:
Since locatedTitle is a shared_ptr, what do you think about this?
```
foundTitles[locatedTitleName] = locatedTitle;
if (!locatedTitle->getDeJureLiege()) // locatedTitle has no de jure liege set yet, which indicated it's newTitle's direct de jure vassal
			locatedTitle->setDeJureLiege(newTitle);
```

Reply:
Much better! I'd be inclined to swap the order of setting the de jure liege and inserting into foundTitles, but its not a huge issue either way

Reply:
✔️